### PR TITLE
Service load balancer contivmodel changes

### PR DESCRIPTION
**Pull request by abhi**

### CODE REVIEW

File: client/contivModelClient.go [hunk 1]
```diff
@@ -349,6 +349,19 @@ type ServiceInstanceLinks struct {
 	Service Link `json:"Service,omitempty"`
 }
 
+type ServiceLB struct {
+	// every object has a key
+	Key string `json:"key,omitempty"`
+
+	IpAddress   string   `json:"ipAddress,omitempty"` // Service ip
+	Labels      []string `json:"labels,omitempty"`
+	Network     string   `json:"network,omitempty"` // Service subnet
```
Comment (jojimt) [inline client/contivModelClient.go]: Why do we need a service subnet?

File: client/contivModelClient.go [hunk 2]
```diff
@@ -349,6 +349,19 @@ type ServiceInstanceLinks struct {
...
+	Ports       []string `json:"ports,omitempty"`
```
Comment (jojimt) [inline client/contivModelClient.go:359]: Does "Ports" encode the protocol as well? Like tcp:80 udp:5550 etc?

File: contivModel.go [hunk 3]
```diff
@@ -296,6 +309,7 @@ type Collections struct {
 	rules            map[string]*Rule
 	services         map[string]*Service
 	serviceInstances map[string]*ServiceInstance
+	ServiceLBs       map[string]*ServiceLB
```
Comment (jojimt) [inline contivModel.go]: Nit: should be serviceLBs to be consistent.

File: contivModel.go [hunk 4]
```diff
@@ -719,8 +756,6 @@ func CreateAppProfile(obj *AppProfile) error {
 		return errors.New("Invalid object type")
 	}
 
-	saveObj := obj
```
Comment (jojimt) [inline contivModel.go:722]: Looks like you don't have the latest modelGen. You're effectively taking out a bugfix (https://github.com/contiv/modelgen/issues/6)

Comment (abhi) [inline contivModel.go:722]: Updated modelgen. Thanks for pointing it out

**Comment by jojimt** [comment]
LGTM

**Comment by jojimt** [comment]
ping @shaleman

File: services.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                                             "type": "string",
+                                             "title": "Service subnet",
+                                             "length": 15
+                                },
+                                "ipAddress":{
+                                            "type":"string",
+                                            "title":"Service ip",
+                                            "length": 15
+                                },
+                                "labels":       {
```
Comment (shaleman) [inline services.json]: selector is probably better than labels

File: services.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                                             "type": "array",
+                                             "title": "labels key value pair",
+                                             "length": 512,
+					     "items" :"string"
+                                },
+                                "ports":{
```
Comment (shaleman) [inline services.json]: Agree with @jojimt's comment. We need to specify protocol as well.

Comment (abhi) [inline services.json]: @shaleman ports is of the format "svcport:provport:protocol" and this can be multiple entries.

File: services.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                                            "type":"array",
+                                            "title":"service provider port",
+                                            "length": 32,
+                                            "items" : "string"
+                                }
+                 }
```
Comment (shaleman) [inline services.json]: Also, please add a link to tenant and network.
Correspondingly:
1. Add a link set in tenant to keep track of all the services
2. Add a link set in network to keep track of all the services in the network.

We should use these to know when a tenant can be deleted, when a service network can be deleted etc.

File: services.json (hunk 5 above)
Comment (abhi) [inline services.json]: @shaleman  cool will change it

File: services.json (hunk 7 above)
Comment (shaleman) [inline services.json]: One more thing, Can you also specify "format:" attribute for all fields.
It'll be good to have field validators going forward..

File: servicelb.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                                "tenantName": {
+                                        "type": "string",
+                                        "title": "Tenant Name",
+                                        "length": 64
+                                },
+
+                                "network": {
+                                             "type": "string",
+                                             "title": "Service subnet",
+                                             "length": 15
```
Comment (shaleman) [inline servicelb.json]: minor: This attribute title should be "Service network name" and length should be 64.

File: servicelb.json [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+					     "items" :"string"
+                                },
+                                "ports":{
+                                            "type":"array",
+                                            "title":"service provider port",
+                                            "length": 32,
+                                            "items" : "string"
+                                }
+                 }
+        }]
```
Comment (shaleman) [inline servicelb.json:51]: Did you forget to add the links from service object?

**Comment by shaleman** [comment]
LGTM other than the minor comments..

**Pull request merged**